### PR TITLE
feat: pii with rebase

### PR DESCRIPTION
**Pull request by skonigs**

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 1]
```diff
@@ -181,3 +183,24 @@ class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsReq
   }
 
 }
+
+object PII extends ComplexParamsReadable[PII]
```
Comment (assafi) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:187]: [Not for this PR] - I'm not sure what the right naming convention, but I think we should rename our objects a bit to include nouns. @mhamilton723 any thoughts?

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDKSchemasV4.scala [hunk 2]
```diff
@@ -149,6 +163,22 @@ object SDKConverters {
       result.getPrimaryLanguage.getWarnings.asScala.toSeq.map(fromSDK))
   }
 
+  implicit def fromSDK(ent: PiiEntity): PIIEntityV4 = {
```
Comment (assafi) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDKSchemasV4.scala:166]: Way to go using `implicit`. Definitely not trivial ;)

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 3]
```diff
@@ -397,4 +403,51 @@ class KeyPhraseExtractionSuiteV4 extends TestBase with DataFrameEquality with Te
...
+    .setLocation("eastus")
+    .setTextCol("text")
+    .setLanguageCol("lang")
+    .setOutputCol("output")
+
+  test("PII - Basic Usage") {
+    val replies = extractor.transform(df)
+      .select("output.result.redactedText")
+      .collect()
+    assert(replies(0).schema(0).name == "redactedText")
```
Comment (assafi) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:445]: Should probably beef up the test a bit to include a few fields. Checking they are not empty for example.

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala (hunk 1 above)
Comment (skonigs) [reply cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:187]: will correct it

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 4]
```diff
@@ -261,24 +250,6 @@ class TextSentimentSuiteV4 extends TestBase with DataFrameEquality with TextKey
 
     assert(errors(0).get(0).toString == "Document text is empty.")
     assert(codes(0).get(0).toString == "InvalidDocument")
-
-    assert(errors(0).get(1).toString == "Document text is empty.")
-    assert(codes(0).get(1).toString == "InvalidDocument")
-  }
-
-  test("Sentiment Analysis - Invalid Language Input") {
```
Comment (assafi) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:269]: Are you sure you want to remove these tests?

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 5]
```diff
@@ -181,3 +180,23 @@ class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsReq
...
+object TextAnalyticsPIIV4 extends ComplexParamsReadable[TextAnalyticsPIIV4]
+
+class TextAnalyticsPIIV4(override val uid: String = randomUID("TextAnalyticsPIIV4"))
+  extends TextAnalyticsSDKBase[PIIEntityCollectionV4]() {
+  logClass()
+
+  override val responseBinding: SparkBindings[TAResponseV4[PIIEntityCollectionV4]]
+  = PIIResponseV4
+
+  override def invokeTextAnalytics(client: TextAnalyticsClient,
```
Comment (assafi) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:193]: It's amazing to see how easy it is now to add these capabilities!

**Pull request merged**